frontend/components/ClubEditPage/AddressField.tsx

```diff
@@ -0,0 +1,110 @@
+import { LoadScript } from '@react-google-maps/api'
+import { Libraries } from '@react-google-maps/api/dist/utils/make-load-script-url'
+import { ReactElement, useEffect } from 'react'
+import CreatbleSelect from 'react-select/creatable'
+import usePlacesAutocomplete from 'use-places-autocomplete'
+
+import { BORDER, CLUBS_GREY, FOCUS_GRAY } from '../../constants/colors'
+
+const styles = {
+  control: ({ background, ...base }) => {
+    return {
+      ...base,
+      border: `1px solid ${BORDER}`,
+      boxShadow: 'none',
+    }
+  },
+  option: ({ background, ...base }, { isFocused, isSelected }) => {
+    const isEmphasized = isFocused || isSelected
+    return {
+      ...base,
+      background: isEmphasized ? FOCUS_GRAY : background,
+      color: CLUBS_GREY,
+    }
+  },
+}
+
+type AddressProps = {
+  addressValue: string
+  changeAddress: (state: any) => void
+}
+
+const AddressTypeaheadField = ({
+  changeAddress,
+  addressValue,
+}: AddressProps): ReactElement => {
+  const {
+    value,
+    suggestions: { data },
+    setValue,
+  } = usePlacesAutocomplete({
+    requestOptions: {
+      location: { lat: () => 39.952104, lng: () => -75.193739 },
+      radius: 4 * 1000,
+    },
+  })
+
+  useEffect(() => {
+    if (addressValue.length > 0) setValue(addressValue)
+  }, [])
+
+  const handleChange = (newValue) => {
+    setValue(newValue)
+    if (newValue) changeAddress(newValue.value)
+  }
+
+  const components = {
+    IndicatorSeparator: () => null,
+    DropdownIndicator: () => null,
+  }
+
+  return (
+    <CreatbleSelect
```

Comment:
Rename this from `CreatbleSelect` to `CreatableSelect`. You should be able to do this easily with F2 on Visual Studio Code.